crates/cli/src/shutdown.rs

```diff
@@ -112,5 +120,11 @@ impl ShutdownManager {
         self.task_tracker().wait().await;
 
         tracing::info!("All tasks are done, exitting");
+
+        if likely_crashed {
+            ExitCode::FAILURE
+        } else {
+            ExitCode::SUCCESS
```

Comment:
I guess that's orthogonal to this PR, but is exiting with SUCCESS really the right thing to do if exiting due to a SIGTERM or so?

It's possible the SIGTERM was sent by another process due to an out of memory situation or something like that, but then a successful exit code doesn't necessarily trigger a restart in e.g. systemd by default. (I've had this failure mode with other services in the past.)

I mostly just wonder what the convention is for this.

Reply:
Hmmm SIGTERM is usually sent when asking for a graceful shutdown. Docker/Kubernetes does that, systemd probably as well